src/main/kotlin/lotto/domain/LottoMachine.kt

```diff
@@ -0,0 +1,26 @@
+package lotto.domain
+
+import lotto.common.LottoTicketPolicy
+
+object LottoMachine {
+    fun calculateTicketCount(amount: Int, manualTicketCount: Int): LottoTicketCount {
+        val autoTicketCount = (amount / LottoTicketPolicy.PRICE) - manualTicketCount
+
+        return LottoTicketCount(autoTicketCount, manualTicketCount)
+    }
+
+    fun process(lottoTicketBundle: LottoTicketBundle, winningBallResult: WinningBallResult): LottoWinningResult {
+        val resultMap = mutableMapOf<LottoTicketResult, Int>()
+        lottoTicketBundle.lottoTickets.groupingBy { ticket ->
+            val intersectNumbers = ticket.intersect(winningBallResult.winningBalls)
+            val isBonusBallMatched = winningBallResult.bonusBall in ticket
+            LottoTicketResult(intersectNumbers.size, isBonusBallMatched)
+        }.eachCountTo(resultMap)
+
+        return LottoWinningResult(resultMap.toSortedMap(getTicketResultComparator()))
```

Comment:
```suggestion
        return lottoTicketBundle.lottoTickets.groupingBy {
            val intersectNumbers = it.numbers.intersect(winningBallResult.winningBalls)
            val isBonusBallMatched = winningBallResult.bonusBall in it.numbers
            LottoTicketResult(intersectNumbers.size, isBonusBallMatched)
        }
            .eachCount()
            .toSortedMap(getTicketResultComparator())
            .run {
                LottoWinningResult(this)
            }
```

이렇게도 표현이 가능해보이네요~ 😊

그리고 LottoTicketResult를 만드는 부분에 대해서 LottoMachine에서 모두 해결하는 것보다는, LottoTicketResult에게 LottoTicketBundle과 WinningBallResult를 넘겨서 좀 더 객체 협력을 만드는 게 좋지 않을까요?